### PR TITLE
Fix pageutils will truncate the file if write failed

### PR DESCRIPTION
**Pull request by jiaqizho**
### What problem does this PR solve?

Issue Number: ref #3594 

Problem Summary:
- If `PageStorage` failed to write, Then PageUtils will truncate it into offset.
- It will make trouble in V3, Because we are perform `random write` in V3 BlobFile.

### What is changed and how it works?
- If write failed. Do not truncate the file into `offset`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by CalvinNeo — APPROVED** [review]
lgtm

File: dbms/src/Storages/Page/PageUtil.h [hunk 1]
```diff
@@ -192,15 +195,21 @@ void writeFile(
...
+
+                int truncate_res = 0;
+                // If write failed in V3, Don't do truncate
+                if (truncate_if_failed)
+                {
+                    // If error occurs, apply `ftruncate` try to truncate the broken bytes we have written.
+                    // Note that the result of this ftruncate is ignored, there is nothing we can do to
+                    // handle ftruncate error. The errno may change after ftruncate called.
+                    truncate_res = ::ftruncate(file->getFd(), offset);
+                }
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/Page/PageUtil.h:207]: This PR keeps the behavior not changed for V2, but disable the truncate after error happen for V3

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**